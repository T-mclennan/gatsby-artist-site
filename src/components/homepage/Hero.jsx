import React from "react"
import { Link } from "gatsby";
import socialLinks from "../../constants/social_links";
import { Parallax } from 'react-scroll-parallax';
import { StaticImage } from "gatsby-plugin-image";
import HeroNav from './HeroNav';
import * as styles from './Hero.module.css';


const Hero = () => {
  return (
    <header className={styles.hero}>
        <HeroNav />
        <div className={styles.heroDetails}>
          <div className={styles.heroText}>
            <h1 className={styles.heroText}>Santa Cruz Street Art</h1>
            <Link to="/about" className="btn center-btn">
              See Gallery
            </Link>
          </div>
        </div>

        <Parallax className="custom-class" y={[-20, 20]} tagOuter="figure">
          <StaticImage 
            // height={1000}
            src="../../assets/images/hallway.jpg"
            alt="header image"
            className="hero-img"
            placeholder="blurred"
          />
        </Parallax>

    </header>
  )
}

export default Hero
