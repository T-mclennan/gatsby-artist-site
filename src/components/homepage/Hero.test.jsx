import React from "react";
import renderer from "react-test-renderer";
import Hero from "./Hero";

describe("Hero", () => {
  it("Has properly rendered", () => {
    const tree = renderer
      .create(<Hero />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})