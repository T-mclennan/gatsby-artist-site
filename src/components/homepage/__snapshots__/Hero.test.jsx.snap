// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Has properly rendered 1`] = `
<section
  style={
    Object {
      "backgroundImage": "url(test-file-stub), linear-gradient(rgba(230, 100, 101, 0.8), rgba(145, 152, 229, 0.8))",
      "height": "100vh",
    }
  }
>
  <nav
    data-sal="fade"
    data-sal-delay="750"
    data-sal-duration="1000"
  >
    <div>
      <div>
        <a
          href="/"
          style={
            Object {
              "marginTop": 5,
            }
          }
        >
          <img
            alt="graff and stuff"
            src="test-file-stub"
          />
        </a>
        <button
          aria-label="menu"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 224h416a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16H16a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16zm416 192H16a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm3.17-384H172.83A12.82 12.82 0 0 0 160 44.83v38.34A12.82 12.82 0 0 0 172.83 96h262.34A12.82 12.82 0 0 0 448 83.17V44.83A12.82 12.82 0 0 0 435.17 32zm0 256H172.83A12.82 12.82 0 0 0 160 300.83v38.34A12.82 12.82 0 0 0 172.83 352h262.34A12.82 12.82 0 0 0 448 339.17v-38.34A12.82 12.82 0 0 0 435.17 288z"
            />
          </svg>
        </button>
      </div>
      <div>
        <a
          href="/about/"
        >
          about
        </a>
        <a
          href="/projects/"
        >
          projects
        </a>
        <a
          href="/gallery/"
        >
          gallery
        </a>
        <a
          href="/music/"
        >
          music
        </a>
        <a
          href="/contact/"
        >
          contact
        </a>
      </div>
    </div>
  </nav>
  <div
    data-sal="slide-up"
    data-sal-delay="300"
    data-sal-duration="1000"
    data-sal-easing="ease"
  >
    <h1>
      Graff and Stuff
    </h1>
    <a
      className="btn center-btn"
      href="/gallery"
    >
      See Gallery
    </a>
  </div>
</section>
`;
